nightfall-deployer/src/circuit-setup.mjs

```diff
@@ -92,8 +93,18 @@ async function setupCircuits() {
   for (let i = 0; i < vks.length; i++) {
     const circuit = circuitsToSetup[i];
 
-    if (!vks[i] || config.ALWAYS_DO_TRUSTED_SETUP) {
-      // we don't have an existing vk so let's generate one
+    const fileBuffer = fs.readFileSync(`./circuits/${circuit}`);
+    const hcircuit = crypto.createHash('md5').update(fileBuffer).digest('hex');
+
+    const checkHash = await axios.post(
```

Comment:
There needs to be a way for force a trusted setup. Also, what happens if ther eis not vk[i] defined?

Reply:
I restored the ability to force trusted setup. 
from the tests I did in the case vk[i] is undefined there was always also a difference on the hash, but for safety I added the vk[i] check, if undefined trusted setup is forced

Reply:
since you already have tests, can you also include those? or were they manual?

Reply:
no, sorry but they were all manual tests